components/script/dom/htmlinputelement.rs

```diff
@@ -292,14 +352,12 @@ impl HTMLInputElementMethods for HTMLInputElement {
 
     // https://html.spec.whatwg.org/multipage/#dom-input-value
     fn Value(&self) -> DOMString {
-        self.textinput.borrow().get_content()
+        InputValueMode::from_input_element(&self).get_value(self)
```

Comment:
I'm not sure how I missed this, but this style is kind of weird. I'd expect `InputValueMode` to be a field in `HTMLInputElement`, but here you're constructing a new `InputValueMode` every time you get or set the value of an `HTMLInputElement`.